### ISSUE TITLE
Browsing with unsupported browser does not show a warning

### ISSUE THREAD

**Issue by tbsbdr** [owncloud/web#7098]
### Steps to reproduce
1. Use Safari 14.x
2. open https://web.cloud.owncloud.com
3. user sees blank page

### Expected behaviour
- Inform the user how to overcome this issue and why she should use the latest version.
- After login, show the "unsupported browser" page (see scribble)
- The user can proceed to ownCloud Web via click on "Proceed at own risk"
- The "unsupported browser" page always shows up after login, if an unsupported browser is detected
- for mobile devices our recommendation is to install the native app
- the "unsupported browser" check can be disabled via config eg. for branded clients / Enterprise Edition

![image](https://user-images.githubusercontent.com/26610733/172597081-e12162b8-c08a-451f-aa57-ece42ef88257.png)

### Actual behaviour
- blank page is shown

**Comment by kulmann** [comment]
> @tbsbdr @kulmann
> We might need a full list of which browsers (with version) we might support, right?

In theory: https://github.com/owncloud/web/blob/550ed2857ef06c65e31610f2bdffe6a61ae83a16/package.json#L32

**Comment by pascalwengerter** [comment]
We originally had some IE deprecation warnings in the console which I moved to the layout topbar (https://github.com/owncloud/web/pull/6629), but I think for this ticket we'd need to build something "static" into the `web-container` (which gets tough a11y-wise since there's no translations for it IIRC)

**Comment by tbsbdr** [comment]
@kulmann I'd suggest the following requirements and wording regarding compatibility.

We could also declare it more dynamic as one might do it nowadays like "Chrome, Firefox, Edge: Latest 2 versions. Safari: Latest version". I decided for the static requirements as it might feel more reliable and specific ex. for corporate decision makers - this is also why I added the operating system column. Might be almost irrelevant but can provide some mental safety.

This would be copied to the documentation as soon as we agree.

\----------------------start----------------------

# Desktop requirements

<table><tbody><tr><td>Browser</td><td>Supported version</td><td>Operating System</td><td>&nbsp;</td></tr><tr><td>Chrome</td><td>104+</td><td>Windows 10+, macOS 12.5+</td><td>&nbsp;</td></tr><tr><td>Firefox</td><td>103+</td><td>Windows 10+, macOS 12.5+</td><td>&nbsp;</td></tr><tr><td>Edge</td><td>104+</td><td>Windows 10+</td><td>&nbsp;</td></tr><tr><td>Safari</td><td>15.6+</td><td>macOS 12.5+</td><td>&nbsp;</td></tr><tr><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td></tr></tbody></table>

*   Touch devices are supported.
*   Minimum resolution is 1280 x 720 px

# Mobile

ownCloud Web provides basic functionality on mobile devices, but not all features are available on mobile devices. For mobile devices, installation of the free Android or iOS App is required:

*   **Android App** [Google Play Store](https://play.google.com/store/apps/details?id=com.owncloud.android)
*   **iOS App** [Apple App Store](https://bit.ly/oCiOSapp)

# Not supported

Other browser such as the Microsoft Internet Explorer are not officially supported.

\-------------------end----------------------

**Comment by kulmann** [comment]
> # Mobile
> ownCloud Web provides basic functionality on mobile devices, but not all features are available on mobile devices. For mobile devices, installation of the free Android or iOS App is required:
> 
> * **Android App** [Google Play Store](https://play.google.com/store/apps/details?id=com.owncloud.android)
> * **iOS App** [Apple App Store](https://bit.ly/oCiOSapp)

For mobile I'd like to use some wording like `For the best mobile experience please use our free Android or iOS App`, what do you think?

I don't feel too good about stating static version numbers. What do you think about the following:

```
## Desktop requirements

We support the latest 2 versions of Chrome, Firefox and Edge. For Safari only the latest version is supported. At the time of writing this (2022-08-22) that results in the following supported versions:
<your table>
```

**Comment by tbsbdr** [comment]
@mmattel as discussed, let's start with something like this for the docs. Feel free to find an unambiguous (precise) but still attractive wording especially for the mobile support for Web (cf. Benedikts comment).

Outlook: I'd be happy if we could dynamically display the latest 2 versions so that visitors don't need to look them up.

# Desktop requirements

<table><tbody><tr><td>Browser</td><td>Supported version</td><td>Operating System</td><td>&nbsp;</td></tr><tr><td>Chrome</td><td>Latest 2 versions</td><td>Windows 10+, macOS 12.5+</td><td>&nbsp;</td></tr><tr><td>Firefox</td><td>Latest 2 versions</td><td>Windows 10+, macOS 12.5+</td><td>&nbsp;</td></tr><tr><td>Edge</td><td>Latest 2 versions</td><td>Windows 10+</td><td>&nbsp;</td></tr><tr><td>Safari</td><td>15.6+</td><td>macOS 12.5+</td><td>&nbsp;</td></tr><tr><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td><td>&nbsp;</td></tr></tbody></table>

*   Touch input is supported.
*   Minimum resolution is 1280 x 720 px

# Mobile

ownCloud Web provides basic functionality on mobile devices, but not all features are available on mobile devices. For mobile devices, installation of the free Android or iOS App is required:

*   **Android App** [Google Play Store](https://play.google.com/store/apps/details?id=com.owncloud.android)
*   **iOS App** [Apple App Store](https://bit.ly/oCiOSapp)

# Not supported

Other browser such as the Microsoft Internet Explorer are not supported.

**Comment by JanAckermann** [comment]
![image](https://user-images.githubusercontent.com/26169327/188619091-a42e4038-ed4f-4bb7-a30f-5b8e4c7464fe.png)

@tbsbdr @kulmann 
Supporting the latest 2 versions for example Chrome has a bad downside. 
Chrome is releasing a new major version every or every other month. 
So showing the BIG 'Browser not supported' notification, as I didn't update the browser in the last 3 months is a no-go for me, also never saw something like this elsewhere in the web. 

I think we need another concept.

**Comment by mmattel** [comment]
Note that the documentation now has a requirement page.
As of now two things are missing in **this** issue:
* desktop --> linux support
* adding ms surface

Those two items are already part of docs

@tbsbdr fyi

**Comment by JanAckermann** [comment]
Ms surface ? Isn’t this a normal windows 10/11 platform, or do they still have arm w10/11?

**Comment by mmattel** [comment]
Technically correct, but pls put yourself into the shoes of an ordinary user. They do not care about the underlaying processor but on the form factor. At that point it is like a mobile but not android or ios. MS Surface stands for a class of devides (like the Lenovo Yoga) but treated as the desktop class...

**Comment by JanAckermann** [comment]
I don’t get it as we develop  responsive web applications

**Comment by dschmidt** [comment]
Fixed in https://github.com/owncloud/web/pull/7942? 

Do we still want to extend the message/add more info?

**Comment by kulmann** [comment]
> Fixed in #7942?
> 
> Do we still want to extend the message/add more info?

Fixed in #7942 😉  Closing.
